### PR TITLE
Logging for link-layer device updates

### PR DESCRIPTION
**Pull request by manadart**
Juju sources networking config from both the machine agent and the provider. These updates result in log lines like this:
```
machine-0: 17:55:27 INFO juju.apiserver.common.networkingcommon processing link-layer devices for machine "0"
```
The problem here is that this indicates neither the model that the machine is in, nor the source of the update. This makes the lines look like duplicates when they are not.

Here we fix the lines so that they do include sufficient detail to indicate the model and the source.

In addition the logging level for address that we skip is changed from `warning` to `info` due to not fulfilling the criteria for warnings - the operator should not raise a bug for them.

## QA steps

- Bootstrap and add a singe machine to the default model.
- Controller log should now differentiate between entries:
```
machine-0: 10:15:04 INFO juju.apiserver.common.networkingcommon processing agent-sourced link-layer devices for machine "0" in model "9ccd40c8-685f-4f16-830f-c26fe71f83eb"
...
machine-0: 10:24:20 INFO juju.apiserver.common.networkingcommon processing agent-sourced link-layer devices for machine "0" in model "c6b2c407-a7ea-4085-8b64-9fbaba01a1af"
machine-0: 10:24:25 INFO juju.apiserver.common.networkingcommon processing provider-sourced link-layer devices for machine "0" in model "c6b2c407-a7ea-4085-8b64-9fbaba01a1af"
```

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1951331

### CODE REVIEW

File: apiserver/common/networkingcommon/networkconfigapi_test.go [hunk 1]
```diff
@@ -247,7 +247,7 @@ func (s *networkConfigSuite) TestUpdateMachineLinkLayerOpMultipleAddressSuccess(
 	s.expectMachine()
 	mExp := s.machine.EXPECT()
 	mExp.AllLinkLayerDevices().Return([]networkingcommon.LinkLayerDevice{lbDev, ethDev, delDev}, nil)
-	mExp.AllAddresses().Return([]networkingcommon.LinkLayerAddress{lbAddr, delAddr}, nil)
```
Comment (manadart) [inline apiserver/common/networkingcommon/networkconfigapi_test.go:250]: This is the result of regenerating mocks. This change was missed back when the method was renamed.

File: state/machine.go [hunk 2]
```diff
@@ -178,12 +178,10 @@ func (m *Machine) ContainerType() instance.ContainerType {
 	return instance.ContainerType(m.doc.ContainerType)
 }
 
-func (m *Machine) ModelName() string {
```
Comment (manadart) [inline state/machine.go:181]: The only use of this was in a test. Replaced and removed to discourage potato programming.

**Review by anvial — APPROVED** [review]
Reproduced QA steps (in my local LXD cloud) where the expected state of the log corresponds to the observed. The code looks properly.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**